packages/tldraw/src/lib/ui/hooks/menu-hooks.ts

```diff
@@ -217,3 +223,137 @@ export function useCanUndo() {
 	const editor = useEditor()
 	return useValue('useCanUndo', () => editor.getCanUndo(), [editor])
 }
+
+/** @public */
+export function useOneEmbeddableBookmarkSelected() {
+	const editor = useEditor()
+	const getEmbedDefinition = useGetEmbedDefinition()
+
+	return useValue(
+		'oneEmbeddableBookmarkSelected',
+		() => {
+			const onlySelectedShape = editor.getOnlySelectedShape()
+			if (!onlySelectedShape) return false
+			return !!(
+				editor.isShapeOfType<TLBookmarkShape>(onlySelectedShape, 'bookmark') &&
+				onlySelectedShape.props.url &&
+				getEmbedDefinition(onlySelectedShape.props.url) &&
+				!editor.isShapeOrAncestorLocked(onlySelectedShape)
+			)
+		},
+		[editor]
+	)
+}
+
+/** @public */
+export function useOneEmbedSelected() {
+	const editor = useEditor()
+
+	return useValue(
+		'oneEmbedSelected',
+		() => {
+			const onlySelectedShape = editor.getOnlySelectedShape()
+			if (!onlySelectedShape) return false
+			return !!(
+				editor.isShapeOfType<TLEmbedShape>(onlySelectedShape, 'embed') &&
+				onlySelectedShape.props.url &&
+				!editor.isShapeOrAncestorLocked(onlySelectedShape)
+			)
+		},
+		[editor]
+	)
+}
+
+/** @public */
+export function useCanFlattenToImage() {
+	const editor = useEditor()
+	return useValue(
+		'should display flatten option',
+		() => {
+			const selectedShapeIds = editor.getSelectedShapeIds()
+			if (selectedShapeIds.length === 0) return false
+			const onlySelectedShape = editor.getOnlySelectedShape()
+			if (onlySelectedShape && editor.isShapeOfType<TLImageShape>(onlySelectedShape, 'image')) {
+				return false
+			}
+			return true
+		},
+		[editor]
+	)
+}
+
+/** @public */
+export function useCanRemoveFrame() {
```

Comment:
some of these feels maybe a little too specific for a particular action to be in this general file...but i don't feel too strongly about it